src/dodal/common/udc_directory_provider.py

```diff
@@ -32,16 +42,13 @@ async def update(self, *, directory: Path, suffix: str = "", **kwargs):
              suffix: Optional str that will be appended to the panda device name along with the file
                 type extension to construct the output filename
         """
-        output_directory = directory / self.resource_dir
-        output_directory.mkdir(exist_ok=True)
```

Comment:
Yeah, we need to make sure the directory is created before the panda PCAP block is armed. As of https://github.com/bluesky/ophyd-async/pull/503/files#diff-71fdf871730c23569996842e011913a3c14590bc0d77bb221c3174246a672d17R63 there is a proper error for this now.

Reply:
Don't lines 52, 56 in the linked change create the directory prior to the check on line 64?

Reply:
Hmmm, that line is confusing. It does:
`await self.panda_data_block.create_directory.set(info.create_dir_depth)`, 
and `info.create_dir_depth` is an `int`. I think it's setting the directory depth, rather than directory itself, despite the name `create_directory`.

I can't see anything here which actually creates a directory

Reply:
I'm not familiar with the PandA HDFWriter, I'm just going off of what I know of the PathProvider and the comments in the file:

```

Reply:
Okay, looking at the panda ioc (https://github.com/PandABlocks/PandABlocks-ioc/blob/4f9247dcc9609593e1eb661eb49373e251047c4b/src/pandablocks_ioc/_hdf_ioc.py#L676), it looks like it _should_ create the directory for us now.

Reply:
> So in your example of /1/2/3/4/5, a value of 2 would mean that we can create any directories starting with index 2, meaning ./3/4/5 can be created if they don't already exist. A value of -2 would mean two from the back, meaning that we could create ./4/5.

From @jwlodek's comment [here](https://github.com/PandABlocks/PandABlocks-ioc/pull/102#issuecomment-2117552752), so -1 is correct to what we want to do: we want to create 1 directory back from the leaf of the path, just the final subdirectory if required. 

Note that this may be a change in behaviour if previously other paths prior to the final subdirectory didn't exist.

Reply:
Right, the panda IOC now handles directory creation, with the directory creation depth determined by the signal written to by the HDF writer, w/ behavior as described in the linked comment. (the behavior is as such because the areaDetector equivalent siganl behaves the same way, and we wanted to be able to use a single path provider for a panda and an AD device.